subprojects/core/src/integTest/groovy/org/gradle/api/tasks/IncrementalTasksIntegrationTest.groovy

```diff
@@ -108,4 +112,70 @@ class FailingIncrementalTasksIntegrationTest extends AbstractIntegrationSpec {
         "remove"      | false       | "with removed outputs is fully rebuilt"
         "none"        | true        | "with unmodified outputs is executed as incremental"
     }
+
+    @Issue("https://github.com/gradle/gradle/issues/9320")
+    def "incremental task with NAME_ONLY input (matching file names and content) detects changed input"() {
+        def taskName = "incrementalTask"
+        def inputs = ('a'..'z').collect { file("${it}/input.txt").createFile() }
+        def modifiedInput = inputs.toList().sort { random() }.first()
+
+        @Language("Groovy") script = """
+            class IncrementalTask extends DefaultTask {
+                @InputFiles
+                @PathSensitive(PathSensitivity.NAME_ONLY)
+                FileCollection inputFiles = project.files(${inputs.collect { "'${it}'" }.join(", ")})
+            
+                @TaskAction
+                def action(IncrementalTaskInputs inputs) {}
```

Comment:
I guess you wanted to add the test to `AbstractIncrementalTasksIntegrationTest`, so it runs for `InputChanges` and not the "old" `IncrementalTaskInputs` interface.

Reply:
The original ticket specified `IncrementalTaskInputs`: https://github.com/gradle/gradle/issues/9320
Which one should I use?

Reply:
We did this together with @paplorinc, and added it for the old API deliberately. We did this because the original issue was reported about the old API. We added it here because it saved us from having to discuss the two APIs we have for incremental changes.

That said, we should add it for the new API as well. Moving it to `AbstractIncrementalTasksIntegrationTest` would accomplish exactly that.

Reply:
Moving it to `AbstractIncrementalTasksIntegrationTest` would have required me to restructure it too much, but added the two types in the where clause.